stan/math/prim/fun/as_array_or_scalar.hpp

```diff
@@ -50,6 +47,19 @@ inline auto as_array_or_scalar(T&& v) {
                      std::forward<T>(v));
 }
 
+/**
+ * Converts a `var_value<T>` with inner Eigen matrix type to an `var_value<T>`
+ *  with an inner array.
+ *
+ * @tparam T Type of `var_value<T>` with inner `Eigen::Matrix` or expression
+ * @param v Specified `var_value<T>` with inner `Eigen::Matrix` or expression.
+ * @return `var_value<>` with Matrix converted to an array.
+ */
+template <typename T, require_var_matrix_t<T>* = nullptr>
+inline auto as_array_or_scalar(T&& v) {
```

Comment:
Yeah I'll move it.

On a side note, one interesting thing I realized the other day is that since all of our signatures are templated now we could actually move pretty much all of `rev/fun` to `prim/fun`. Since SFINAE happens before the body is evaluated the function in `rev` that are templated will never be instantiated without also include `var/is_var`